File: mvp_app/MyProject/libtsparser/src/main/java/com/tosmart/xiexuming/libtsparser/table/Component.java
package com.tosmart.xiexuming.libtsparser.table;

import androidx.annotation.NonNull;

import com.tosmart.xiexuming.libtsparser.descriptor.Descriptor;

import java.util.ArrayList;

/**
 * @author xiexuming
 */
public class Component {
    private int streamType;
    private int elementaryPid;
    private int elementStreamInfoLength;
    private ArrayList<Descriptor> descriptor;

    @NonNull
    @Override
    public String toString() {
        return "Component{" +
                "streamType=" + streamType +
                ", elementaryPID=" + elementaryPid +
                ", ESInfoLength=" + elementStreamInfoLength +
                ", descriptor=" + descriptor +
                '}';
    }

    public int getStreamType() {
        return streamType;
    }

    public void setStreamType(int streamType) {
        this.streamType = streamType;
    }

    public int getElementaryPid() {
        return elementaryPid;
    }

    public void setElementaryPid(int elementaryPid) {
        this.elementaryPid = elementaryPid;
    }

    public int getElementStreamInfoLength() {
        return elementStreamInfoLength;
    }

    public void setElementStreamInfoLength(int elementStreamInfoLength) {
        this.elementStreamInfoLength = elementStreamInfoLength;
    }

    public ArrayList<Descriptor> getDescriptor() {
        return descriptor;
    }

    public void setDescriptor(ArrayList<Descriptor> descriptor) {
        this.descriptor = descriptor;
    }
}

File: newpaserapp/libtosmartparse/src/main/java/com/example/libtosmartparse/table/Component.kt
package com.example.libtosmartparse.table

import com.example.libtosmartparse.descriptor.Descriptor

/**
 * @author xxm
 * 2022/7/17
 **/
class Component {
    private var streamType = 0
    private var elementaryPid = 0
    private var elementStreamInfoLength = 0
    private var descriptor: ArrayList<Descriptor>? = null
    override fun toString(): String {
        return "Component{" +
                "streamType=" + streamType +
                ", elementaryPID=" + elementaryPid +
                ", ESInfoLength=" + elementStreamInfoLength +
                ", descriptor=" + descriptor +
                '}'
    }

    fun getStreamType(): Int {
        return streamType
    }

    fun setStreamType(streamType: Int) {
        this.streamType = streamType
    }

    fun getElementaryPid(): Int {
        return elementaryPid
    }

    fun setElementaryPid(elementaryPid: Int) {
        this.elementaryPid = elementaryPid
    }

    fun getElementStreamInfoLength(): Int {
        return elementStreamInfoLength
    }

    fun setElementStreamInfoLength(elementStreamInfoLength: Int) {
        this.elementStreamInfoLength = elementStreamInfoLength
    }

    fun getDescriptor(): ArrayList<Descriptor>? {
        return descriptor
    }

    fun setDescriptor(descriptor: ArrayList<Descriptor>?) {
        this.descriptor = descriptor
    }
}